### PR TITLE
Bugfix sort encoding

### LINKED ISSUE #59: df126b8 breaks REMIND input data preparation
[log-19638235.log](https://github.com/pik-piam/madrat/files/6494858/log-19638235.log)

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
Non-utf8 strings no longer lead to crashes (fixes #59) and sorting is now locale independent everywhere.
Added the internal robustOrder and robustSort functions, which are wrapper functions reound base::order and base::sort, but they convert character vectors to utf8 and always use local independent radix sort. In madrat these functions should be used instead of base::order and base::sort.

### CODE REVIEW

File: R/robustOrder.R [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#' @param decreasing If TRUE decreasing/descending order, if FALSE increasing/ascending order. For the "radix" method,
+#' this can be a vector of length equal to the number of arguments in ... . For the other methods, it must be length one.
+#' @param method Default is "radix", which is locale independent. The alternatives "auto" and "shell" should not be used
+#' in madrat because they are locale dependent.
+#' @seealso \code{\link[base]{order}}
+#' @author Pascal Führlich
+robustOrder <- function(..., na.last = TRUE, decreasing = FALSE, method = "radix") {
+  args <- list(...)
+  if (any(lapply(args, length) == 0)) {
+    return()
```
Comment (tscheypidi) [inline R/robustOrder.R:19]: I don't get the meaning of that part. The function should have no return value if any of the arguments provided has the length 0?

File: R/robustOrder.R [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  args <- lapply(args, sapply, function(element) {
+    ifelse(is.character(element), enc2utf8(element), element)
```
Comment (tscheypidi) [inline R/robustOrder.R:22]: I've read that the `ifelse` function is rather slow. I would suggest to change it to:
```suggestion
    if(is.character(element)) element <- enc2utf8(element)
    return(element)
```

**Review by tscheypidi — CHANGES_REQUESTED** [review]
looks good, just one suggestion and one question for clarification

File: R/robustOrder.R (hunk 1 above)
Comment (pfuehrlich-pik) [reply R/robustOrder.R:19]: This check prevents an error when an empty vector is sorted. `return()` still returns `NULL`, which works, but to better match the behavior of base::order I changed it to return vector(mode = "integer", length = 0).

**Pull request merged**